benchmarks/timeseries/periodic_feature_extractor.py

```diff
@@ -0,0 +1,235 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+Benchmarks for the PeriodicFeatureExtractor class.
+
+This module contains benchmarks that are comparing
+the performance of a numpy implementation with a python
+implementation.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from datetime import datetime, timedelta, timezone
+from functools import partial
+from timeit import timeit
+from typing import List
+
+import numpy as np
+from frequenz.channels import Broadcast
+from numpy.random import default_rng
+from numpy.typing import NDArray
+
+from frequenz.sdk.timeseries import MovingWindow, PeriodicFeatureExtractor, Sample
+
+
+async def init_feature_extractor(period: int) -> PeriodicFeatureExtractor:
+    """Initialize the PeriodicFeatureExtractor class."""
+    # We only need the moving window to initialize the PeriodicFeatureExtractor class.
+    lm_chan = Broadcast[Sample]("lm_net_power")
+    # monkey patch the MovingWindow class to have a fixed size such that
+    # it can be initialized with a period bigger then the window size.
+    MovingWindow.__len__ = lambda self: period  # type: ignore[method-assign]
```

Comment:
The patching is still there!

Reply:
weird I can't find it

Reply:
ah damn found it. That was done in 2 places.